addon/bootstrap.js

```diff
@@ -114,17 +104,15 @@ function handleStartup() {
     resourceURI: addonResourceURI
   });
 
-  maybeClearAddonCache().then(() => {
-    if (!shouldDisable() && !webExtension.started) {
-      start(webExtension);
-    } else if (shouldDisable()) {
-      stop(webExtension);
-    }
-  });
+  if (!shouldDisable() && !webExtension.started) {
+    start(webExtension);
+  } else if (shouldDisable()) {
+    stop(webExtension, ADDON_DISABLE);
```

Comment:
I'm not sure if it's better to pass a fake reason to shutdown, or just not pass a reason at all. I *guess* we can just do this, for now?